### PR TITLE
SOE-3465: Fix signup block placement

### PR DESCRIPTION
**Pull request by minorwm**
# READY FOR REVIEW

# Summary
- Push Email Subscribe block back up on Mag Article node (except on Collection Node)

# Needed By (Date)
- ?

# Urgency
- N/A

# Steps to Test

1. Go to: `magazine/article/new-ai-camera-recognizes-objects-faster-and-more-efficiently` and `magazine/article/dan-boneh-still-early-days-blockchain-rich-possibility`.
2. See the email signup block above the footer on both pages.
3. Checkout this branch.
4. `drush cc all`
5. Go to: `magazine/article/new-ai-camera-recognizes-objects-faster-and-more-efficiently`.
6. The email subscribe block should now be under "Related Departments" and above "Related Articles".
7. Go to: `magazine/article/dan-boneh-still-early-days-blockchain-rich-possibility`
8. The email subscribe block should be under "Related Articles" as this article is part of a collection.

# Affected Projects or Products
- Engineering
- `stanford_soe_helper`

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SOE-3465

## Folks to notify
@boznik 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: stanford_soe_helper.module [hunk 1]
```diff
@@ -79,6 +79,25 @@ user/*/edit";
       ];
     }
   }
+
+  $collections = 0;
+
+  // If the page is part of a collection, take note.
+  if (isset($variables['page']['content']['system_main']['nodes'])) {
+    foreach ($variables['page']['content']['system_main']['nodes'] as $node) {
```
[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module:87]: Ignoring this.

[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module:87]: Ignoring this.

File: stanford_soe_helper.module [hunk 2]
```diff
@@ -79,6 +79,28 @@ user/*/edit";
...
+        $collections = TRUE;
+      }
+    }
+  }
+
+  // If we are on a mag article that is not part of a collection.
+  if (!$collections) {
+    if (isset($variables['page']['content_bottom']) && isset($variables['page']['fullwidth_bottom']['bean_stanford-soe-mag-news-signup'])) {
+      // Move the mag news signup between related depts and related articles.
+      $signup = $variables['page']['fullwidth_bottom']['bean_stanford-soe-mag-news-signup'];
```
[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module]: Ignoring this.

File: stanford_soe_helper.module [hunk 3]
```diff
@@ -79,6 +79,28 @@ user/*/edit";
...
+      $variables['page']['content_bottom'] = array('bean_stanford-soe-mag-news-signup' => $signup) + $variables['page']['content_bottom'];
```
[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module]: Ignoring this.

File: stanford_soe_helper.module [hunk 4]
```diff
@@ -79,6 +79,23 @@ user/*/edit";
...
+  foreach ($variables['page']['content']['system_main']['nodes'] as $node) {
+    if (isset($node['field_s_mag_article_collection'])) {
+      $collections++;
+    }
+  }
+
+  // Move the mag news signup between related depts and related articles.
+  if (!$collections) {
+    if (isset($variables['page']['fullwidth_bottom']['bean_stanford-soe-mag-news-signup'])) {
+      array_unshift($variables['page']['content_bottom'], $variables['page']['fullwidth_bottom']);
```
[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module:98]: Ignoring this.

[the remark this answers is not in the document]
Comment (minorwm) [reply stanford_soe_helper.module:98]: Ignoring this.

**Review by boznik — APPROVED** [review]
working as expected

**Pull request merged**